Decompressing lz4 kernel image failed

Device:  LG G6 h870
Android version: 9.0 PIE
Magisk version name: Canary
Magisk version code: 24101

Last working build 23.0
Tested not working 24.0, 24.1, and latest canary

[Magisk 23.0 log](https://github.com/topjohnwu/Magisk/files/7962742/magisk_install_log_2022-01-29T021905Z.log)
[Magisk 24.0 log](https://github.com/topjohnwu/Magisk/files/7962744/magisk_install_log_2022-01-29T022607.log)

23.0 RAMDISK_SZ      [5343737]
24.0 RAMDISK_SZ      [5524638]

The latest canary output is 9 MB while the 23.0 is around 23 MB
[Magisk patched 23.0 working boot.img](https://github.com/topjohnwu/Magisk/files/7962749/magisk_patched-23000_RloK5.zip)
[Magisk patched 24001 not working](https://github.com/topjohnwu/Magisk/files/7962750/magisk_patched-24001_PzwBx.zip)

Hello, Is there any way to compile the older versions of magisk?
The newest version compiles without errors, but when i checkout the 68ac409bfd0f9ea176ae33ed055be950ccb4b384 there are errors like:
```
* Building the stub app
Error:
        Attribute application@appComponentFactory value=(s.eX) from AndroidManifest.xml:12:9-45
        is also present at AndroidManifest.xml:12:9-45 value=(yJ.CJH).
```
```
* Building binaries: magisk magiskinit magiskboot busybox
make: *** No rule to make target 'jni/external/selinux/libsepol/src/deprecated_funcs.c', needed by 'obj/local/armeabi-v7a/objs/sepol/selinux/libsepol/src/depreca
ted_funcs.o'.  Stop.
```

--- a/native/jni/magiskboot/compress.cpp
+++ b/native/jni/magiskboot/compress.cpp
@@ -520,7 +520,7 @@ protected:
             return false;
         }
         if (bwrite(&block_sz, sizeof(block_sz)) && bwrite(out_buf, block_sz)) {
-            in_total += sizeof(block_sz) + block_sz;
+            in_total += len;
             return true;
         }
         return false;

